sympy/functions/elementary/tests/test_trigonometric.py

```diff
@@ -1356,6 +1357,21 @@ def test_asec():
     assert asec(x).rewrite(acsc) == -acsc(x) + pi/2
 
 
+def test_asec_is_real():
+    assert asec(S.Zero).is_real is None
```

Comment:
If you really want to test your assumptions implementation, probably this is the way to go.
```
z = Symbol('z', zero=True)
test asec(z).is_real
```